packages/happy-dom/src/nodes/html-select-element/HTMLSelectElement.ts

```diff
@@ -28,10 +28,20 @@ export default class HTMLSelectElement extends HTMLElement implements IHTMLSelec
 	public onchange: (event: Event) => void | null = null;
 	public oninput: (event: Event) => void | null = null;
 
-	public _value = null;
-	public _selectedIndex = -1;
 	public _options: IHTMLOptionsCollection = null;
 
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		super();
+
+		// Clear our options cache when option elements are added/removed from the DOM, keeping it fresh.
+		new MutationObserver(() => {
```

Comment:
Option elements must be children of the Select element, so instead of using a MutationObserver we can just override `appendChild()`, `removeChild()` and `insertBefore()` to know which option elements it contains.

When elements are added or removed, then I think we should add and remove them as an index on the Select element as well (e.g. `this[0] = addedOption`);. We need to add the "length" property. After doing that we can easily use the select itself as an array (e.g. `for(let i = 0; i < this.length; i++) {  console.log(this[0]);  }`).